Intro to GTFS

machikoyasuda opened the pull request:
closes #614 

## What this PR does
- Creates a new page at `https://www.camobilitymarketplace.org/intro-gtfs-rt`
- Upgrades MobiMart to the latest Bootstrap, one minor version

## Code approach
- Use utility classes for styles as much as possible
- Uses Bootstrap variables for the `warning` and `info` colors for this box pattern:
![image](https://github.com/user-attachments/assets/c50b9b37-d8be-4d63-8e16-5c673f92f30c)

machikoyasuda commented on src/_sass/_base.scss:
```diff
@@ -155,6 +152,6 @@ hr {
-  font-size: calc(14rem / 16);
+.bg-info-beige {
+  background-color: $beige;
 }
```
Moved `fs-7` class to the Utilites file with the rest of the font size classes.

machikoyasuda commented on src/_sass/_base.scss:
```diff
@@ -47,10 +47,6 @@ figure {
-  text-decoration: none !important;
-}
-
 p {
```
It's not recommended to set a style on all links like this, with an `!important` too. At this point, it was being individually overridden everywhere. Removed.

machikoyasuda commented on src/_layouts/default.html:
```diff
@@ -15,12 +15,11 @@
     <link
-      href="https://cdn.jsdelivr.net/npm/bootstrap@5.2.3/dist/css/bootstrap.min.css"
+      href="https://cdn.jsdelivr.net/npm/bootstrap@5.3.3/dist/css/bootstrap.min.css"
       rel="stylesheet"
-      integrity="sha384-rbsA2VBKQhggwzxH7pPCaAqO46MgnOM80zW1RWuH61DGLwZJEdK2Kadq2F9CUG65"
+      integrity="sha384-QWTKZyjpPEjISv5WaRU9OFeRpok6YctnYmDr5pNlyT2bRjXh0JMhjY6hW+ALEwIH"
```
Upgraded to latest Bootstrap to get more background color classes, which are used in the GTFS Schedule/Realtime boxes.

machikoyasuda commented on src/_sass/_variables.scss:
```diff
@@ -42,10 +44,16 @@ $custom-select-indicator: url('data:image/svg+xml,<svg width="12" height="7" vie
+  --bs-link-color-rgb: #{$primary-rgb};
   --bs-body-font-family: "Poppins", sans-serif;
   --bs-link-hover-color: #{$dark-blue};
+  --bs-info-bg-subtle: #d5eef5;
+  --bs-info-rgb: 161, 216, 233; // border
+  --bs-warning-bg-subtle: #fff1d0;
+  --bs-warning-rgb: 254, 223, 149; // border
```
These are used for the background and border colors of these boxes:
![image](https://github.com/user-attachments/assets/438b57a9-c25a-4400-aef2-6e2a27eab129)

machikoyasuda commented:
Now ready for review. 

This PR will be merged first (without being linked anywhere), then https://github.com/cal-itp/mobility-marketplace/pull/617 can be updated with the link to `/intro-gtfs-rt`, and then finally, after these 2 PRs are merged, the last nav one can be merged: https://github.com/cal-itp/mobility-marketplace/pull/618

allejo commented:
@thekaveman yes I can 👀

machikoyasuda commented:
@ohmegasquared Ready for design review https://deploy-preview-619--cal-itp-mobility-marketplace.netlify.app/intro-gtfs-rt

thekaveman commented on src/intro-gtfs-rt.html:
```diff
@@ -0,0 +1,218 @@
+      </li>
+      <li>
+        Learn more about what constitutes high quality GTFS data
+        <a href="https://dot.ca.gov/cal-itp/california-minimum-general-transit-feed-specification-gtfs-guidelines-v2_0"
```
This should be the latest minimum guidelines link, the same one used above, right?

Here: https://github.com/cal-itp/mobility-marketplace/pull/619/files#diff-ef26ea713f21540506ea7b0fe43519c55548742e705edd1c3f7593678eef0591R152

`https://dot.ca.gov/cal-itp/california-transit-data-guidelines`

thekaveman replied:
Yeah let's go ahead and update to the 4.0 link in both places. 

FYI @leelas-pm @ohmegasquared

leelas-pm replied:
Thank you! Sorry for not catching that!

machikoyasuda replied:
@leelas-pm 

Should the Resources section have a link to the outdated "minimum guidelines" still? Should I take it out? 
<img width="780" alt="image" src="https://github.com/user-attachments/assets/37ba125f-7639-430b-9627-3ba5ed76e6c9" />
(Personally I lean towards taking it out if it's outdated)

leelas-pm replied:
Yes! Definitely take it out - thanks for the ping!

thekaveman replied:
Sorry, I didn't realize it was a dupe!

leelas-pm replied:
Adding reviewing for duplicates and outdated links to my QA process for website work! 🙇

allejo reviewed (changes requested):
Since the majority of this was written with atomic classes, we could achieve the same thing almost entirely with atomic classes. I think the only new CSS that should be add is what you have for the icons.

Also, I hope you don't mind that I pushed a commit of slightly smaller versions of the images. I just ran them through ImageOptim

machikoyasuda commented:
@allejo Thank you for making all those commit suggestions!

allejo commented on src/intro-gtfs-rt.html:
```diff
@@ -0,0 +1,230 @@
+    <div class="row gx-2 my-3">
+      <div class="col-md-6 mb-3 mb-md-0">
+        <div class="rounded rounded-3 border border-info bg-info-subtle p-3 h-100">
+          <h3 class="h4 with-gtfs-schedule d-flex align-items-center">GTFS Schedule</h3>
```
```suggestion
          <h3 class="h4 with-gtfs-schedule d-flex align-items-center">GTFS Schedule Data</h3>
```

My only remaining concern is the duplicate `h3` headings for the little boxes under the "So, what exactly is GTFS?" heading.

| preview | doc structure |
|-|-|	
| ![image](https://github.com/user-attachments/assets/3e17f001-d75b-4408-866d-89888f810ba7) | ![image](https://github.com/user-attachments/assets/3ebd3c89-276a-4bfd-afd8-376c9cf3f1c6) |

thekaveman replied:
If we're going this route (I think it's a good catch and I agree) -- the `GTFS Schedule` heading in the second set of boxes should be updated as well: `GTFS Schedule Data`

machikoyasuda replied:
Hrm... What should the H3 to? Or just.. not use Headings for the GTFS Schedule/GTFS Realtime boxes at all?

allejo replied:
As long as this heading is changed to include "data," then all the h3 headings will be unique and no further changes are needed.

If you want to move away from using an h3, then we could probably use a `<p>` tag and tie it to the `<ul>` with an `aria-labelledby`. But I think the h3 is fine, it just needs to include "data" so that it's not conflicting with duplicates in the same section

thekaveman replied:
Yeah it's the repeated heading text that is a problem, right? So just changing the text of the second heading fixes it.

machikoyasuda replied:
OOOH~~~ I only just noticed the different `Data` name on Figma 😅 😅 

Thanks for the careful eyes and the commit suggestions Vlad! Resolved. Phew!

ohmegasquared commented:
Its looking great! I have some small feedback items  

### Contracts  Page 
1. missing deep link to the new GTFS page  

<img width="709" alt="Screenshot 2025-03-06 at 11 10 37 AM" src="https://github.com/user-attachments/assets/d9dc561d-aaa9-423f-a70b-c51cfb334d6e" />

### Contracts Detail Page 
2. On the filtered view, we are spelling out RT in the tag and the table 
<img width="181" alt="Screenshot 2025-03-06 at 11 12 32 AM" src="https://github.com/user-attachments/assets/fc113935-09ef-4372-8eae-2b04d9108225" />
<img width="266" alt="Screenshot 2025-03-06 at 11 13 13 AM" src="https://github.com/user-attachments/assets/407f3450-2b51-49eb-992e-29f6574722de" />

3. The link should be mailto:hello@calitp.org. Perhaps we could add a subject line "Question about contracts"
<img width="422" alt="Screenshot 2025-03-06 at 12 39 09 PM" src="https://github.com/user-attachments/assets/e83a939e-c86c-46ec-9ae7-5bc56de00a05" />


### GTFS-RT How To guide
4. The TOC is cut off for both me and Leela, it might be a common viewing screen 
wondering if we should a) make the header hide on scroll or b) make the TOC scrollable? or another light weight option?
<img width="1467" alt="Screenshot 2025-03-06 at 11 40 32 AM" src="https://github.com/user-attachments/assets/e65577cc-df3c-466c-9150-7158aae9f0cd" />

5. Missing links and not using the paragraph link style (paragraph text with underline) 
<img width="1177" alt="Screenshot 2025-03-06 at 12 12 47 PM" src="https://github.com/user-attachments/assets/23f6dd06-5080-407c-a4e5-04d82a6be6bd" />

Missing in: How do I get started?
- GTFS-Realtime MSA Information Packet 
- GTFS-Realtime Introduction
Missing in: Draft a Scope of Work (SOW)
- GTFS-Realtime MSA Information Packet 
- GTFS-Realtime Introduction

Wrong link:
- Draft Scope Of Work > MSA SOW Template and Guidance link is https://docs.google.com/document/d/1ZywIdKwYaL5U3ScmSyfov_N9eEXaEDt3/edit
- Send the SOW to vendors > MSA SOW Template and Guidance link is https://docs.google.com/document/d/1ZywIdKwYaL5U3ScmSyfov_N9eEXaEDt3/edit
- STD 213 should link to https://www.dgs.ca.gov/ols/Forms?search=213&topicCategoryFilters=&audienceCategoryFilters=9048d318e266493abfd2361dc40d1f45&sort=relevance&activeFilters=audience%7C&page=1 not the form 

6. Links should open in a new tab, so users don't lose their place in this multi-step process (as recommended in https://theadminbar.com/accessibility-weekly/opening-links-in-new-tabs-or-not/) 
- interest form
- GTFS-Realtime MSA Information Packet 
- GTFS-Realtime Introduction
- California GTFS Quality Dashboard
- details on data plans
- available MSAs
- MSA SOW Template And Guidance
- STD 213 
- Sample User Agreement
- California GTFS Quality Dashboard

Will make another comment for intro to GTFS-RT page

machikoyasuda commented:
Hey @ohmegasquared This PR is ONLY for this page: https://deploy-preview-619--cal-itp-mobility-marketplace.netlify.app/intro-gtfs-rt

The How To page is on a different PR and different URL entirely: https://deploy-preview-617--cal-itp-mobility-marketplace.netlify.app/how-to/gtfs-rt (Note that this URL uses 617 in the URL, not 619. These numbers correspond to the PR number).

<img width="1512" alt="image" src="https://github.com/user-attachments/assets/eee7fcfc-60ca-49d4-adf1-d3747e80d6d9" />

allejo reviewed (approved):
Looking good to me 😄

ohmegasquared commented:
PS - mobile looks great!

machikoyasuda commented:
Final Thursday status update:
- Code is approved ✔️ by Vlad
- Design is ready for final review by @ohmegasquared 

Notes:
- I fixed all the incorrect or missing links (2 email links, 1 reports link missing)
- I made the By the numbers section headings black
- I made all links open in a new tab
- I noticed the Resources header was too small, so I fixed that.
- One change from a few days ago: As flagged by @thekaveman and approved by @leelas-pm, we are no longer linking to the Minimum Guidelines https://github.com/cal-itp/mobility-marketplace/pull/619#discussion_r1980593799 - which is linked twice on the comps: once in the GTFS quality section and once in the Resources section.

ohmegasquared commented:
Not seeing the link from contracts to this page in this PR, not sure if its supposed to be somewhere else. 
<img width="634" alt="Screenshot 2025-03-07 at 4 58 13 PM" src="https://github.com/user-attachments/assets/7ae44dda-8d05-4edf-b4a5-928610498cfc" />

I wasn't aware that we decided to remove the minimum guidelines link. It doesn't make sense to keep the text about minimum guidelines and not link to it. Will sync with @leelas-pm if we should remove the references in full

machikoyasuda commented:
@ohmegasquared Ready for final review

ohmegasquared reviewed (approved):
Looks good!
